bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeAirDevice.java

```diff
@@ -0,0 +1,530 @@
+
+            // save the outcome
+            setIsBound(true);
+        } catch (IOException e) {
+            throw new GreeException("Unable to bind to device", e);
+        }
+    }
+
+    public void setDevicePower(DatagramSocket clientSocket, int value) throws GreeException {
+        setCommandValue(clientSocket, GREE_PROP_POWER, value);
+    }
+
+    public void SetDeviceMode(DatagramSocket clientSocket, int value) throws GreeException {
+        if ((value < 0 || value > 4)) {
+            throw new GreeException("Device mode out of range!");
+        }
+        setCommandValue(clientSocket, GREE_PROP_MODE, value);
+    }
+
+    public void setDeviceSwingUpDown(DatagramSocket clientSocket, int value) throws GreeException {
+        // Only values 0,1,2,3,4,5,6,10,11 allowed
+        if ((value < 0 || value > 11) || (value > 6 && value < 10)) {
+            throw new GreeException("SwingUpDown value out of range!");
+        }
+        setCommandValue(clientSocket, GREE_PROP_SWINGUPDOWN, value);
+    }
+
+    public void setDeviceSwingLeftRight(DatagramSocket clientSocket, int value) throws GreeException {
+        setCommandValue(clientSocket, GREE_PROP_SWINGLEFTRIGHT, value, 0, 6);
+    }
+
+    /**
+     * Only allow this to happen if this device has been bound and values are valid
+     * Possible values are :
+     * 0 : Auto
+     * 1 : Low
+     * 2 : Medium Low
+     * 3 : Medium
+     * 4 : Medium High
+     * 5 : High
+     */
+    public void setDeviceWindspeed(DatagramSocket clientSocket, int value) throws GreeException {
+        if (value < 0 || value > 5) {
+            throw new GreeException("Value out of range!");
+        }
+
+        HashMap<String, Integer> parameters = new HashMap<>();
+        parameters.put(GREE_PROP_WINDSPEED, value);
+        parameters.put(GREE_PROP_QUIET, 0);
+        parameters.put(GREE_PROP_TURBO, 0);
+        parameters.put(GREE_PROP_NOISE, 0);
+        executeCommand(clientSocket, parameters);
+    }
+
+    public void setDeviceTurbo(DatagramSocket clientSocket, int value) throws GreeException {
+        setCommandValue(clientSocket, GREE_PROP_TURBO, value, 0, 1);
+    }
+
+    public void setQuietMode(DatagramSocket clientSocket, int value) throws GreeException {
+        setCommandValue(clientSocket, GREE_PROP_QUIET, value, 0, 2);
+    }
+
+    public int getDeviceTurbo() {
+        return getIntStatusVal(GREE_PROP_TURBO);
+    }
+
+    public void setDeviceLight(DatagramSocket clientSocket, int value) throws GreeException {
+        setCommandValue(clientSocket, GREE_PROP_LIGHT, value);
+    }
+
+    /**
+     * @param value set temperature in degrees Celsius or Fahrenheit
+     */
+    public void setDeviceTempSet(DatagramSocket clientSocket, QuantityType<?> temp) throws GreeException {
+        // If commanding Fahrenheit set halfStep to 1 or 0 to tell the A/C which F integer
+        // temperature to use as celsius alone is ambigious
+        Double newVal = temp.doubleValue();
```

Comment:
Use primitives if possible.
```suggestion
        double newVal = temp.doubleValue();
```

Reply:
done